internal/statemachine/helper.go

```diff
@@ -354,3 +371,150 @@ func maxOffset(offset1, offset2 quantity.Offset) quantity.Offset {
 	}
 	return offset2
 }
+
+// createPartitionTable creates a disk image file and writes the partition table to it
+func createPartitionTable(volumeName string, volume *gadget.Volume, sectorSize uint64, isSeeded bool) *partition.Table {
+	var gptPartitions = make([]*gpt.Partition, 0)
+	var mbrPartitions = make([]*mbr.Partition, 0)
+	var partitionTable partition.Table
+
+	for _, structure := range volume.Structure {
+		if structure.Role == "mbr" || structure.Type == "bare" ||
+			shouldSkipStructure(structure, isSeeded) {
+			continue
+		}
+
+		var structureType string
+		// Check for hybrid MBR/GPT
+		if strings.Contains(structure.Type, ",") {
+			types := strings.Split(structure.Type, ",")
+			if volume.Schema == "gpt" {
+				structureType = types[1]
+			} else {
+				structureType = types[0]
+			}
+		} else {
+			structureType = structure.Type
+		}
+
+		if volume.Schema == "mbr" {
+			bootable := false
+			if structure.Role == gadget.SystemBoot || structure.Label == gadget.SystemBoot {
+				bootable = true
+			}
+			// mbr.Type is a byte. snapd has already verified that this string
+			// is exactly two chars, so we can parse those two chars to a byte
+			partitionType, _ := strconv.ParseUint(structureType, 16, 8)
+			mbrPartition := &mbr.Partition{
+				Start:    uint32(*structure.Offset) / uint32(sectorSize),
```

Comment:
One small inconsistency I see here is that for the legacy ubuntu-image we used `ceil(offset / self.sector_size)` (same for size). It shouldn't be a big of a deal, but I think *just in case* we should do a similar thing here. Since right now this is basically discarding the reminder, so like `floor()`.

Reply:
I have added math.Ceil to all of these calculations.